spec/deqm_test_kit/data_requirements_spec.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+RSpec.describe DEQMTestKit::DataRequirements do
+  let(:suite) { Inferno::Repositories::TestSuites.new.find('deqm_test_suite') }
+  let(:group) { suite.groups[3] }
+  let(:test_session) { repo_create(:test_session, test_suite_id: 'deqm_test_suite') }
+  let(:url) { 'http://example.com/fhir' }
+  # ensure this url matches url in embedded_client in data_requirements.rb
+  let(:embedded_client) do
+    'http://cqf_ruler:8080/cqf-ruler-r4/fhir'
+  end
+  let(:error_outcome) { FHIR::OperationOutcome.new(issue: [{ severity: 'error' }]) }
+
+  def run(runnable, inputs = {})
+    test_run_params = { test_session_id: test_session.id }.merge(runnable.reference_hash)
+    test_run = Inferno::Repositories::TestRuns.new.create(test_run_params)
+    Inferno::TestRunner.new(test_session: test_session, test_run: test_run).run(runnable, inputs)
+    Inferno::Repositories::TestRuns.new.results_for_test_run(test_run.id)
+  end
+
+  describe 'data requirements test' do
+    let(:test) { group.tests.first }
+    let(:measure_name) { 'EXM130' }
+    let(:measure_version) { '7.3.000' }
+    let(:test_id) { 'test_id' }
+
+    it 'passes if a Library was received' do
+      test_measure_response = FHIR::Bundle.new(total: 1, entry: [{ resource: { id: test_id } }])
+      test_library_response = FHIR::Library.new(dataRequirement: [{ type: 'Condition' }])
+
+      stub_request(:get, "#{url}/Measure?name=#{measure_name}&version=#{measure_version}")
+        .to_return(status: 200, body: test_measure_response.to_json)
+
+      stub_request(:get, "#{embedded_client}/Measure?name=#{measure_name}&version=#{measure_version}")
+        .to_return(status: 200, body: test_measure_response.to_json)
+
+      stub_request(:post, "#{url}/Measure/#{test_id}/$data-requirements")
```

Comment:
Can someone explain why the data-requirements operation ends up being a POST request? I thought data-requirements was supposed to be a GET, but the outgoing request from fhir_operation is always a POST

Reply:
The inferno-core operation implementation just default [does a post](https://github.com/inferno-community/inferno-core/blob/c4081927733881e3624c24eb78959f7ca06daa65/lib/inferno/dsl/fhir_client.rb#L80). Inferno team did say it was a bit immature. You're right that this operation would typically be a GET (https://www.hl7.org/fhir/measure-operation-data-requirements.html). I wonder if this has the potential to trip up non-cqf_ruler servers.